src/Helpers/LeanCode.UserIdExtractors/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,91 @@
+using LeanCode.DomainModels.Ids;
+using LeanCode.UserIdExtractors.Extractors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace LeanCode.UserIdExtractors;
+
+public static class UserServiceProviderExtensions
```

Comment:
This class is IMO slightly over engineered. Why not provide three different methods, one for Guid, one for string, and one for typed ids? You could then get rid of the reflection.

Reply:
To get rid of the reflection we would probably need 4 methods: guid, string, raw and prefixed because `IRawTypedId` has 2 generic arguments: `TBacking` and `TSelf` and `TBacking` would also have to be provided as generic argument to the extension method.

So we could:
* Keep it as it is,
* Have 3 methods: non-generic string/guid, generic typedId for prefixed and raw (but uses reflection),
* Have 4 methods: non-generic string/guid, generic prefixed (with 1 arg and proper generic constraints), generic raw (with 2 args and proper generic constraints).